packages/svelte/src/compiler/phases/1-parse/state/tag.js

```diff
@@ -263,38 +265,53 @@ function open(parser) {
 		return;
 	}
 
-	if (parser.eat('snippet')) {
-		parser.require_whitespace();
-
-		const name_start = parser.index;
-		const name = parser.read_identifier();
-		const name_end = parser.index;
+	if (parser.match('snippet')) {
+		const snippet_declaraion_end = find_matching_bracket(parser.template, parser.index, '{');
```

Comment:
```suggestion
		const snippet_declaration_end = find_matching_bracket(parser.template, parser.index, '{');
```

Reply:
come to think of it, can't we use the existing `read_context` (which should really be called `read_pattern`) logic here, just for multiple arguments instead of one?

Reply:
Long story short, I think no. We were using it before and I couldn't find a way to make it work in this case. I believe this is also somewhat more efficient (and also benefits const block parsing), but I'm not positive.